lib/createdb.sh

```diff
@@ -394,7 +403,11 @@ if [ "$createdb" = y ]; then
     eval $MYSQLADMIN $mycreatedb_args $myargs $FLAGS --default-character-set=utf8 create $DBNAME || exit 1
 fi
 
-allhosts="localhost 127.0.0.1 localhost.localdomain$granthosts"
+if $allow_all_hosts; then
+    allhosts="%"
```

Comment:
MySQL 8.4 documentation: “The host name part of the account name, if omitted, defaults to '%'. You should be aware that, while MySQL 8.4 treats grants made to such a user as though they had been granted to 'user'@'localhost', this behavior is deprecated, and thus subject to removal in a future version of MySQL.” So there would need to be something fancier here.